firedrake/utility_meshes.py

```diff
@@ -512,6 +513,21 @@ def UnitTriangleMesh(
     plex = mesh.plex_from_cell_list(
         2, cells, coords, comm, mesh._generate_default_mesh_topology_name(name)
     )
+
+    # mark boundary facets
+    plex.createLabel(dmcommon.FACE_SETS_LABEL)
+    plex.markBoundaryFaces("boundary_faces")
+    coords = plex.getCoordinates()
+    coord_sec = plex.getCoordinateSection()
+    labels = [1,2,0] # Faces are y=0, x+y=0, x=0.
+    boundary_faces = plex.getStratumIS("boundary_faces", 1).getIndices()
+    for face, label in zip(boundary_faces, labels):
+        plex.setLabelValue(dmcommon.FACE_SETS_LABEL, face, label)
+    plex.removeLabel("boundary_faces")
```

Comment:
To be consistent with `UnitSquareMesh` and friends, and also inspired by FIAT facet ordering, we probably want to use `1` (diagonal), `2`(x=0), `3`(y=0).
It is also safer to label facets using coordinates as done in `TensorRectangleMesh`.

Reply:
Sounds fair. Added in last commit.